Test failure with pyparsing < 2.2.1

`pytest` tests fail with `pyparsing< 2.2.1`, but no such requirement is specified anywhere AFAIK.  This is particularly notable as the Debian stable ("buster") ships with 2.2.0.  The reason for the failures is the `DeprecationWarning`s as noted in: https://github.com/pyparsing/pyparsing/issues/12

`pyparsing` is not a direct requirement of `pytest` currently, but `packaging` _is_ and that pulls it in.

`pip list` output before installing `pyparsing==2.2.1`:
```
Package             Version  
------------------- ---------
alabaster           0.7.8    
apipkg              1.5      
appdirs             1.4.3    
argcomplete         1.8.1    
argon2-cffi         18.3.0   
asn1crypto          0.24.0   
atomicwrites        1.1.5    
attrs               18.2.0   
Babel               2.6.0    
beautifulsoup4      4.7.1    
bleach              3.1.2    
blinker             1.4      
certifi             2018.8.24
Chameleon           2.24     
chardet             3.0.4    
check-manifest      0.37     
configobj           5.0.6    
coverage            4.5.2    
croniter            0.3.24   
cryptography        2.6.1    
defusedxml          0.5.0    
distro              1.3.0    
docutils            0.14     
elementpath         2.0.4    
entrypoints         0.3      
execnet             1.5.0    
filelock            3.0.10   
html5lib            1.0.1    
hupper              1.5      
hypothesis          3.71.11  
idna                2.6      
imagesize           1.0.0    
importlib-metadata  1.6.0    
iniconfig           1.1.1    
iotop               0.6      
itsdangerous        0.24     
Jinja2              2.10     
jsonpatch           1.21     
jsonpointer         1.10     
jsonschema          2.6.0    
keyring             17.1.1   
keyrings.alt        3.1.1    
lxml                4.3.2    
Mako                1.0.7    
MarkupSafe          1.1.0    
memory-profiler     0.52.0   
mock                2.0.0    
more-itertools      4.2.0    
msgpack             0.5.6    
nose                1.3.7    
oauthlib            2.1.0    
olefile             0.46     
packaging           19.0     
passlib             1.7.1    
Paste               3.0.6    
PasteDeploy         2.0.1    
PasteScript         2.0.2    
pbr                 4.2.0    
Pillow              5.4.1    
pip                 18.1     
pkg-resources       0.0.0    
pkginfo             1.4.2    
plaster             1.0      
plaster-pastedeploy 0.5      
pluggy              0.13.1   
psutil              5.5.1    
py                  1.9.0    
pycrypto            2.6.1    
pycryptodomex       3.6.1    
Pygments            2.3.1    
PyGObject           3.30.4   
PyJWT               1.7.0    
pyOpenSSL           19.0.0   
pyparsing           2.2.0    
pyramid             1.10.2   
python-apt          1.8.4.1  
python-dateutil     2.7.3    
pytz                2019.1   
pyxdg               0.25     
PyYAML              3.13     
pyzmq               17.1.2   
readme-renderer     24.0     
repoze.lru          0.7      
requests            2.21.0   
requests-toolbelt   0.8.0    
roman               2.0.0    
ruamel.yaml         0.15.34  
salt                3002     
SecretStorage       2.3.1    
setuptools          40.8.0   
setuptools-scm      3.2.0    
simplejson          3.16.0   
six                 1.12.0   
soupsieve           1.8      
Sphinx              1.8.4    
sphinx-rtd-theme    0.4.3    
systemd-python      234      
Tempita             0.5.2    
toml                0.10.0   
tox                 3.7.0    
tqdm                4.28.1   
translationstring   1.3      
twine               1.13.0   
urllib3             1.24.1   
venusian            1.2.0    
virtualenv          15.1.0   
waitress            1.2.0b2  
webencodings        0.5.1    
WebOb               1.8.5    
WebTest             2.0.32   
wheel               0.32.3   
Whoosh              2.7.4    
xmlschema           1.2.5    
zipp                1.0.0    
zope.component      4.3.0    
zope.deprecation    4.4.0    
zope.event          4.2.0    
zope.hookable       4.0.4    
zope.interface      4.3.2 
```

Tests failing with `pyparsing==2.2.0` (above `pip list`):
```
                  FAILED testing/io/test_terminalwriter.py::test_code_highlight[with markup and code_highlight]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[2-missing]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[2-missing-1-ok]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[1-ok]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[1-ok-pin-exact]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[1-ok-pin-loose]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[missing-version]
                  FAILED testing/test_config.py::TestParseIni::test_missing_required_plugins[missing-versions]
                  FAILED testing/test_terminal.py::test_color_yes - Failed: fnmatch: '\x1b[[]1m...
                  FAILED testing/test_terminal.py::TestCodeHighlight::test_code_highlight_simple
                  FAILED testing/test_terminal.py::TestCodeHighlight::test_code_highlight_continuation
```

.. and the same with `pyparsing==2.2.1`.  Note that failing tests are from a lack of `pygments>2.4.2` and are unrelated.  I understand that this is still an optional requirement of sorts.
```
                  FAILED testing/io/test_terminalwriter.py::test_code_highlight[with markup and code_highlight]
                  FAILED testing/test_terminal.py::test_color_yes - Failed: fnmatch: '\x1b[[]1m...
                  FAILED testing/test_terminal.py::TestCodeHighlight::test_code_highlight_simple
                  FAILED testing/test_terminal.py::TestCodeHighlight::test_code_highlight_continuation
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,6 +23,8 @@ xfail_strict = true
 filterwarnings = [
     "error",
     "default:Using or importing the ABCs:DeprecationWarning:unittest2.*",
+    # produced by older pyparsing<=2.2.0.
+    "default:Using or importing the ABCs:DeprecationWarning:pyparsing.*",
     "default:the imp module is deprecated in favour of importlib:DeprecationWarning:nose.*",
     "ignore:Module already imported so cannot be rewritten:pytest.PytestWarning",
     # produced by python3.6/site.py itself (3.6.7 on Travis, could not trigger it with 3.6.8)."

